### ISSUE TITLE
How to update the icon of search alias?

### ISSUE THREAD

**Issue by chenerg** [brookhong/Surfingkeys#1752]
I have configured with the following script, 
```
api.removeSearchAlias('b');
api.addSearchAlias('b', 'bilibili', 'https://search.bilibili.com/all?keyword=', 's', 'https://s.search.bilibili.com/main/suggest?func=suggest&suggest_type=accurate&sub_type=tag&main_ver=v1&highlight=&upuser_num=3&term=?', function(response) {
    var res = JSON.parse(response.text).result.tag;
    return res.map(function(r){
        return r.value;
    });
   }, 
   {favicon_url: 'https://www.bilibili.com/favicon.ico'}
);
```
however, the icon is still be the icon of baidu, how can I update the icon to the bilibili website?

**Comment by al3xfischer** [comment]
I just encountered the same problem.

I did some research and had a look at the implementation (introduced by #1657).
Which works well if we use a new search alias (for example 'x'). 
Because 'x'  is no predefined alias.
If we try to remove an existing and alias and replace it with some other, the icon is already saved in the local storage.
As they are loaded by default.
And since the stored icon is preferred over the `favicon_url` we are not able to use the `favicon_url`.

https://github.com/brookhong/Surfingkeys/blob/c9ab83e9f6277d2c5fbf04e3126c0b89555b038a/src/content_scripts/ui/omnibar.js#L1254-L1273

IMO this behaviour could be fixed by either changing the implementation of the configurable favicon to favour the `favicon_url` or by removing the local store entry in the `removeSearchAlias`.
One downside of the second approach is that we have to manage domain specific removals.

**Comment by keyboardset** [comment]
hi, how could we remove icon, I do not like icon and want to switch it back to words, is there anyway I could do that？ Thanks for your help.

**Comment by brookhong** [comment]
Should have been fixed in 1.16.0, ahead of which the alias letter was used the key of the icon, so if you are overriding a search alias, the existing icon won't be updated.

Since 1.16.0, the name of search engine (`bilibili` in this case) is being used as the key, the icon will be updated since `bilibili` is a different string from `baidu`.

And you can follow the instructions from @tywhisky to clear icons from application storage.

**Comment by Fred-Vatin** [comment]
One thing to know to make favicon_url update: don’t use space in your search engine name (`prompt` parameter in `addSearchAlias`).
